File: test/benchmark/python/instantiation.py
# This benchmark stresses instance creation and initializer calling.
import time


class Foo:
    def __init__(self):
        pass


start = time.process_time()
i = 0
while i < 500000:
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    Foo()
    i = i + 1

print(time.process_time() - start)

File: test/benchmark/ruby/instantiation.rb
# This benchmark stresses instance creation and initializer calling.
class Foo
    #Python fields are always public whereas Ruby's are private. attr_accessor makes them public
    attr_accessor
    def initialize()
    end
end


start = Time.now
i = 0
while i < 500000
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    Foo.new
    i = i + 1
end

p(Time.now - start)
